src/services/contentProgress.js

```diff
@@ -306,14 +314,36 @@ async function setStartedOrCompletedStatus(contentId, collection, isCompleted) {
 
   const hierarchy = await getHierarchy(contentId, collection)
 
-  await Promise.all([
-    db.contentProgress.recordProgressesTentative(trickleProgress(hierarchy, contentId, collection, progress), collection),
-    bubbleProgress(hierarchy, contentId, collection).then(bubbledProgresses => db.contentProgress.recordProgressesTentative(bubbledProgresses, collection))
-  ])
+  let ids = {}
+  ids = {
+    ...trickleProgress(hierarchy, contentId, collection, progress),
+    ...await bubbleProgress(hierarchy, contentId, collection)
+  }
+  await db.contentProgress.recordProgressesTentative(ids, collection)
+
+  if (collection && collection.type === COLLECTION_TYPE.LEARNING_PATH) {
+    let exportIds = ids
+    exportIds[contentId] = progress
+    await duplicateLearningPathProgressToExternalContents(exportIds, collection, hierarchy)
+  }
 
   return response
 }
 
+async function duplicateLearningPathProgressToExternalContents(ids, collection, hierarchy) {
+  // filter out LPs. we dont want to duplicate to LP's while we dont have a-la-cart LP's set up.
+  const filteredIds = Object.fromEntries(
+    Object.entries(ids).filter((id) => {
+      return hierarchy.parents[parseInt(id)] !== null
+    })
+  )
+
+  // each handles its own bubbling.
+  Object.entries(filteredIds).forEach(([id, pct]) => {
+    saveContentProgress(parseInt(id), null, pct)
```

Comment:
Unless you want fire-and-forget, you may need **await** here since this method is async like the other persist funcs.